pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -818,6 +818,7 @@ func (c *Controller) updateEDS(ep *v1.Endpoints, event model.Event) {
 						UID:             uid,
 						ServiceAccount:  kubeToIstioServiceAccount(pod.Spec.ServiceAccountName, pod.GetNamespace()),
 						Network:         c.endpointNetwork(ea.IP),
+						Locality:        c.GetPodLocality(pod),
```

Comment:
Only one small concern, this function is not efficient,  the node labels is the second choice, if you could calculate it when necessary it is better.

https://github.com/istio/istio/blob/cf272a2b447617976645d50a62e3c5f8247ebe14/pilot/pkg/serviceregistry/kube/controller.go#L322-L338

Reply:
Good point that would be more efficient